readthedocs/projects/views/private.py

```diff
@@ -907,98 +907,86 @@ class EnvironmentVariableDelete(EnvironmentVariableMixin, DeleteView):
-            }
-        )
-
-    download_data = request.GET.get('download', False)
+    template_name = 'projects/projects_search_analytics.html'
+    http_method_names = ['get']
 
-    # if the user has requested to download all data
-    # return csv file in response.
-    if download_data:
-        return _search_analytics_csv_data(request, project_slug)
+    def get(self, request, *args, **kwargs):
+        download_data = request.GET.get('download', False)
+        if download_data:
+            return self._search_analytics_csv_data()
+        return super().get(request, *args, **kwargs)
 
-    # data for plotting the line-chart
-    query_count_of_1_month = SearchQuery.generate_queries_count_of_one_month(
-        project_slug
-    )
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        project = self.get_project()
 
-    queries = []
-    qs = SearchQuery.objects.filter(project=project)
-    if qs.exists():
-        qs = (
-            qs.values('query')
-            .annotate(count=Count('id'))
-            .order_by('-count', 'query')
-            .values_list('query', 'count')
+        context['show_analytics'] = project.has_feature(
+            Feature.SEARCH_ANALYTICS,
         )
+        if not context['show_analytics']:
+            return context
 
-        # only show top 100 queries
-        queries = qs[:100]
-
-    return render(
-        request,
-        'projects/projects_search_analytics.html',
-        {
-            'project': project,
-            'queries': queries,
-            'show_analytics': True,
-            'query_count_of_1_month': query_count_of_1_month,
-        }
-    )
+        # data for plotting the line-chart
+        query_count_of_1_month = SearchQuery.generate_queries_count_of_one_month(
+            project.slug,
+        )
 
+        queries = []
+        qs = SearchQuery.objects.filter(project=project)
+        if qs.exists():
+            qs = (
+                qs.values('query')
+                .annotate(count=Count('id'))
+                .order_by('-count', 'query')
+                .values_list('query', 'count')
+            )
 
-def _search_analytics_csv_data(request, project_slug):
-    """Generate raw csv data of search queries."""
-    project = get_object_or_404(
-        Project.objects.for_admin_user(request.user),
-        slug=project_slug,
-    )
+            # only show top 100 queries
+            queries = qs[:100]
 
-    now = timezone.now().date()
-    last_3_month = now - timezone.timedelta(days=90)
+        context.update(
+            {
+                'queries': queries,
+                'query_count_of_1_month': query_count_of_1_month,
+            },
+        )
+        return context
 
-    data = (
-        SearchQuery.objects.filter(
-            project=project,
-            created__date__gte=last_3_month,
-            created__date__lte=now,
+    def _search_analytics_csv_data(self):
```

Comment:
I think this could be better as a helper/util function returning the data instead a of a class method and be called from the class that constructs the response.

Not really relevant at this point since it's not called from another place, though.